protocol/java/org/openyolo/protocol/Credential.java

```diff
@@ -247,27 +247,30 @@ public void writeToParcel(Parcel dest, int flags) {
          * Starts the process of creating a credential, based on the properties of the
          * provided protocol buffer representation of the credential. The protocol buffer
          * must contain valid data for a credential.
+         * @throws MalformedDataException if the given protocol buffer is invalid.
          */
-        private Builder(@NonNull Protobufs.Credential proto) {
-            // required properties
-            setIdentifier(proto.getId());
-            setAuthenticationMethod(
-                    AuthenticationMethodConverters.PROTOBUF_TO_OBJECT
-                            .convert(proto.getAuthMethod()));
-            setAuthenticationDomain(
-                    AuthenticationDomainConverters.PROTOBUF_TO_OBJECT
-                            .convert(proto.getAuthDomain()));
-
-            // optional properties
-
-            setIdToken(proto.getIdToken());
-            setDisplayName(proto.getDisplayName());
-            setDisplayPicture(proto.getDisplayPictureUri());
-            setPassword(proto.getPassword());
-            setAdditionalProperties(
-                    CollectionConverter.convertMapValues(
-                            proto.getAdditionalPropsMap(),
-                            ByteStringConverters.BYTE_STRING_TO_BYTE_ARRAY));
+        private Builder(@NonNull Protobufs.Credential proto) throws MalformedDataException {
+            try {
+                // required properties
+                setIdentifier(proto.getId());
+                setAuthenticationMethod(
+                        AuthenticationMethodConverters.PROTOBUF_TO_OBJECT
+                                .convert(proto.getAuthMethod()));
+                setAuthenticationDomain(AuthenticationDomain.fromProtobuf(proto.getAuthDomain()));
+
+                // optional properties
+
+                setIdToken(proto.getIdToken());
+                setDisplayName(proto.getDisplayName());
+                setDisplayPicture(proto.getDisplayPictureUri());
+                setPassword(proto.getPassword());
+                setAdditionalProperties(
+                        CollectionConverter.convertMapValues(
+                                proto.getAdditionalPropsMap(),
+                                ByteStringConverters.BYTE_STRING_TO_BYTE_ARRAY));
+            } catch (IllegalArgumentException ex) {
```

Comment:
it might be better to organize this such that the fromProtobuf method catches the IllegalArgumentException and translates to a MalformedDataException.